docs/src/proposals/append-vec-storage.md

```diff
@@ -0,0 +1,374 @@
+offset of the next blob account meta.
+
+#### Account Pubkey Block
+The account pubkey is accessed via the `local_id` of the `StoredAccountMetaV2`,
+which is derived from the location of one account meta when an AppendVec is
+first opened (i.e., it's location inside the vector).  Specifically, the account
+pubkey of one account is located at
+
+```
+account_pubkey_block_offset + 32 + local_id * ACCOUNT_PUBKEY_SIZE
+```
+
+Again, as the offset of each block is aligned.  A 32-byte hash entry means each
+entry is naturally 64-bit aligned without padding.
+
+```
++--------------------------------------+
+| Hash (32 bytes for the entire block) |
++--------------------------------------+
+| account pubkey 1                     |
++--------------------------------------+
+| account pubkey 2                     |
++--------------------------------------+
+|         ...                          |
++--------------------------------------+
+| account pubkey N                     |
++--------------------------------------+
+```
+
+```
++-------------------------------------------------------------+
+| Account Pubkey (32 bytes each)                              |
++-------------------------------------------------------------+
+| pubkey          | 32 bytes                                  |
++-------------------------------------------------------------+
+```
+
+#### Owners Block
+The owners block stores the owner's information.  An owner entry can be accessed
+via `owner_block_offset` + 32 + `owner_local_id` * `OWNER_ENTRY_SIZE`.
+
+Again, as the offset of each block is aligned.  A 32-byte hash entry means each
+entry is naturally 64-bit aligned without padding.
+
+Note that the existing 8-byte `write_version` field in `StoredMeta` is removed
+in this design as we now have AccountsCache as a write cache so that we don't
+need to maintain the `write_version` to enable multiple AppendVec per slot.
+
+```
++--------------------------------------+
+| Hash (32 bytes for the entire block) |
++--------------------------------------+
+| owner 1                              |
++--------------------------------------+
+| owner 2                              |
++--------------------------------------+
+|         ...                          |
++--------------------------------------+
+| owner N                              |
++--------------------------------------+
+```
+
+```
++-------------------------------------------------------------+
+| Owner (32 bytes each)                                       |
++-------------------------------------------------------------+
+| pubkey          | 32 bytes                                  |
++-------------------------------------------------------------+
+```
+
+Again, as mentioned earlier in the Header section, the Owners Block also
+offers forward compability as long as the newer version only adds new fields
+and does not change the definition of the existing fields.
+
+#### (Small) Account Data Blocks
```

Comment:
The current doc here is a bit outdated as I am currently prototyping.

In the current prototype, the account data blocks will be the first thing in the file, so the first block is guaranteed to be page-aligned. But since each block is compressed, the rest of the blocks will not be page-aligned.

Let me think if we could get benefits from both compress and page alignment.